munit/shared/src/main/scala/munit/FileLayout.scala

```diff
@@ -0,0 +1,75 @@
+package munit
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.StandardOpenOption
+import java.nio.file.Path
+
+trait FileLayout {
+
+  protected def fileLayoutDefaultTempDirName = "file-layout"
+
+  /**
+   * Parse a file layout from string
+   *
+   * @param layout
+   * @return the file layout as a map of path -> content
+   * @throws IllegalArgumentException if the layout is formally invalid
+   */
+  protected def parseFileLayout(layout: String): Map[String, String] =
```

Comment:
Should this be a public method? We don't have an equivalent method to this in `scala.meta.testkit.StringFS` https://github.com/scalameta/scalameta/blob/3d66f6c561d26886dda508aff0c314d65761df68/scalameta/testkit/src/main/scala/scala/meta/testkit/StringFS.scala

Instead, we have something like `parse(path: Path): String` which recursively walks a directory and renders a layout.

Reply:
Are you proposing to introduce `parse(path: Path): String` and making `parse(layout: String): Map[String, String]` private?

I think parsing the layout is useful when looking for specific patterns in a file. Recursively walking the directory will pick-up all the files, which may not be needed and be unnecessarily expensive.

For reference, I'm using it in  
https://github.com/buildo/retro/blob/master/tapiro/core/src/test/scala/io/buildo/tapiro/TapiroSuite.scala#L127-L131

Reply:
We can keep `parse(layout: String): Map[String, String]` public but how about renaming it into `parseLayout`? That leaves it open to later implement `parsePath`